### ISSUE TITLE
Negative wind volumes in imported models

### ISSUE THREAD

**Issue by jhmatthews** [sirocco-rt/sirocco#918]
I'm working with a qwind imported model, and I found some very strange behaviour - segmentation faults and negative wind luminosities in some cells. The issue seems to be that some of the cells have negative volumes - if I add a check in cylindrical, for example, as folows: 

```C
      cell_volume = 2 * PI * (rmax * rmax - rmin * rmin) * (zmax - zmin);

      if (cell_volume < 0)
      {
        Error ("negative volume! %8.4e %8.4e %8.4e %8.4e %8.4e\n", cell_volume, rmax, rmin, zmax, zmin);
      }
```

I get outputs like the following: 

```
negative volume! -1.1894e+49 rmax 1.9609e+16 rmin 1.6970e+16 zmax 0.0000e+00 zmin 1.9609e+16
negative volume! -8.2768e+47 rmax  0.0000e+00 rmin1.9609e+16 zmax 3.4258e+14 zmin 0.0000e+00
```

which happens ~60 times in total and seems to correspond to some edge of the wind. The problem is that zmax and rmax can be 0 and less than rmin. these values are set from the wind_z and wind_x arrays, so I'm guessing these arrays aren't quite set up right, either in terms of the final elements, or the number of elements we are looping over.  It's entirely possible I've messed up the imported model - but in which case it should be diagnosable. 

**More generally,** do we have a generic routine that checks wind volumes for both plausibility and to check that inwind and volume variables match ok? If not, can we add one? The code shouldn't be able to run ionization cycles and/or fail through segfaults, it should quit and warn the user. 

To reproduce, run the attached pf file with the attached 30x30 grid.
[qwind_30x30_rmax1.7e16.pf.txt](https://github.com/agnwinds/python/files/7561928/qwind_30x30_rmax1.7e16.pf.txt)
[grid_qwind_30x30_rmax1.7e16.txt](https://github.com/agnwinds/python/files/7561929/grid_qwind_30x30_rmax1.7e16.txt)

**Comment by jhmatthews** [comment]
So I think this is actually due to setting up the ghost cells wrong. I'll fix that in the imported model - but I think the general point about checking the model more carefully is worth doing something about. This model shouldn't be able to run in the first place because it gives nonsense results in certain cells.

**Comment by kslong** [comment]
This should be something that can be added in a moderately straightforward fashion, except that one would need to define the requirements for imported models.  Looking at your file (which I know you said had something in the wrong order), it was not completely obvious what was wrong.

**Comment by jhmatthews** [comment]
Proposed solution - just add some critical error messages in create_maps. This is a reasonably logical place to do it in terms of the program since it will catch errors in both imported and non imported models. Alternatively we could have a separate routine to do this, but this seems as good a place as any. 

```C
int
create_maps ()
{
  int i, j;
  j = 0;

  for (i = 0; i < NDIM2; i++)
  {
    wmain[i].nwind = i;
    if (wmain[i].vol > 0)
    {
      wmain[i].nplasma = j;
      plasmamain[j].nplasma = j;
      plasmamain[j].nwind = i;
      j++;
      if (wmain[i].inwind < 0)
      {
        Error
          ("create_maps: wind cell %d (nplasma %d) has volume but not flagged as in wind! Critical error, could cause undefined behaviour. Exiting.\n", i, j);
        Exit (0);
      }
    }
    else
    {
      wmain[i].nplasma = NPLASMA;
      if (wmain[i].inwind >= 0)
      {
        Error
          ("create_maps: wind cell %d has zero volume but flagged inwind! Critical error, could cause undefined behaviour. Exiting.\n", i);
        Exit (0);
      }
    }
  }
  if (j != NPLASMA)
  {
    Error ("create_maps: Problems with matching cells -- Expected %d Got %d\n", NPLASMA, j);
    Exit (0);
  }

  plasmamain[NPLASMA].nplasma = NPLASMA;
  plasmamain[NPLASMA].nwind = -1;
  return (0);
}
```

**Comment by jhmatthews** [comment]
We could also probably have better checks in the imported wind models for the boundary cells being set up correctly.

**Comment by jhmatthews** [comment]
Is my fix in #922 sufficient here? If so it can be closed.

**Comment by kslong** [comment]
It looks sufficient to me and so I will close the issue.
